dev/grid.html

```diff
@@ -20,20 +20,18 @@
 
       const grid = document.querySelector('vaadin-grid');
 
-      grid.dataProvider = ({ parentItem, page, pageSize }, cb) => {
-        // Let's have 100 root-level items and 5 items on every child level
+      grid.dataProvider = ({ parentItem, page, pageSize }, callback) => {
         const levelSize = parentItem ? 5 : 100;
 
-        const pageItems = [...Array(Math.min(levelSize, pageSize))].map((_, i) => {
-          const indexInLevel = page * pageSize + i;
-
+        const items = Array.from({ length: levelSize }, (_, i) => {
```

Comment:
Hmm....isn't this approach generating a lot of excess items? Let's say you have a grid with 100000 root items, then each data provider call would create an array with 100000 items but only use a small slice of it.

Reply:
Good point, maybe for `grid.html` which can be used to test large grids, the old approach is better.